app/jobs/update_developer_response_rate_job.rb

```diff
@@ -17,8 +15,11 @@ def perform(developer)
   private
 
   def eligable_conversations(developer)
+    grace_period = Rails.application.config.developer_response_grace_period
+    grace_period_start = grace_period.nil? ? Time.current : grace_period.ago
```

Comment:
I'm a little nervous about this conditional because it means the only time `grace_period.ago` is called is in production.

Do you have any ideas on how we could test that codepath?

Reply:
Also, we might be able to simplify this a bit with:

grace_period = Rails.application.config.developer_response_grace_period || 0.seconds
conversation.created_at > grace_period.ago && !conversation.developer_replied?

Reply:
> we might be able to simplify this a bit with

Nice! I've changed that.

> The only time grace_period.ago is called is in production.

The `developer_response_grace_period` is set to `nil` only in development, not during the tests. That's why I set the config to `nil` in [seeds](https://github.com/joemasilotti/railsdevs.com/blob/d824ddbd084c5d076ab7b19f78a304f4babfdf67/db/seeds.rb#L12-L13) because there are some smoke tests in Github Action that were [failing](https://github.com/joemasilotti/railsdevs.com/actions/runs/4495813460/jobs/7909850263).

Now I'm thinking... If the the grace period was set to `0.seconds` in development only because of the seeds, now it can be any value (0+ seconds). 

Should I change the grace period in development? What value should I use?

Reply:
Done!